encoding/resampling.md

```diff
@@ -1,35 +1,429 @@
 # Resampling
 
-## Upsampling
+Resampling is a technique applied in various
+image processing tasks,
+most prominently scaling/resizing,
+but also shifting and rotation.
 
-Or supersampling, upscaling. It is important to know which
-resamplers work best for upsampling and to be able to tell (to some
-extent) what sort of resizing kernel was used on a source in order to
-reverse bad upscales.
+## Resizing
 
-<https://diff.pics/AOJ4sBhbe6SX/1> - 2x magnified luma test image using
-various kernels (and nnedi3)
+The most common class of resampling algorithms
+used for resizing are the convolution-based resamplers.
+As the name suggests,
+these work by [convolving][Convolution] the image
+with a filter function.
+These filter functions,
+also known as kernels,
+are what the terms Bilinear/Bicubic/Lanczos/etc. generally refer to.
+The [ImageWorsener documentation][ImageWorsener] features a great
+visual explanation of how this process works in detail.
+It is strongly recommended to read it.
+If you wish to explore the mathematics behind the design
+of these filters,
+the [Sampling and Reconstruction chapter][Sampling chapter]
+of [Physically Based Rendering][PBRT] is one of the best places
+for a layman to start.
 
-- 3 - B-Spline - Bicubic b=1, c=0
-- 4 - Hermite - Bicubic b=0, c=0
-- 5 - Mitchell-Netravali - Bicubic b=1/3, c=1/3
-- 6 - Catmull-Rom - Bicubic b=0, c=0.5
-- 7 - Sharp Bicubic - Bicubic b=0, c=1
+[Convolution]: https://en.wikipedia.org/wiki/Convolution
+[Sampling chapter]: https://www.pbrt.org/chapters/pbrt_chapter7.pdf
+[PBRT]: https://www.pbrt.org/
+[ImageWorsener]: https://entropymine.com/imageworsener/resample/
 
+### Filters
 
-##  Downsampling
+All resampling kernels behave slightly differently
+and generate artifacts of differing kinds and severity.
 
-TODO
+It should be noted that there is no “objectively best” resampling filter,
+so it is largely a matter of personal preference.
+There are no hard and fast rules about
+which resampler performs best for any given type of content,
+so it’s best to test for yourself.
 
+A short overview of the most common filters follows.
+For a much more extensive explanation of the different filters,
+including details on the exact algorithms,
+see [ImageMagick’s guide][IM].
 
-##  Inverse scaling
+[IM]: http://www.imagemagick.org/Usage/filter/
 
-TODO
+##### Box filter / Nearest Neigbour
```

Comment:
Should be h4 because the last heading, Filters, is h3. Also applies to the following.